### PR TITLE
Clean radium apps

### PR DESCRIPTION
**Pull request by nmellado**
Rename apps names, improve doc, remove useless example

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# All dot files, if you wish to track one, just add it
+.*
+
+# Misc
+*Thumbs.db
+
+# Output Directories
+Bundle*/
+build*/
+cmake-build-*/
```
Comment (MathiasPaulin) [inline .gitignore:20]: missing new line at the end of file ...
```suggestion
cmake-build-*/

```

File: CLISubdivider/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+Excluding command parsing, only very few steps are required to load, simplify and save the object:
+
+ 1. Load triangular mesh or generate 
+```cpp
+Ra::Core::Geometry::TriangleMesh mesh;
+Ra::IO::OBJFileManager obj;
+
+// Load geometry as triangle
+if ( a.inputFilename.empty() ) { mesh = Ra::Core::Geometry::makeBox(); }
+else                           { obj.load( a.inputFilename, mesh ); }
```
Comment (MathiasPaulin) [inline CLISubdivider/README.md:28]: what is `a`?

File: CLISubdivider/README.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+Ra::Core::Geometry::TopologicalMesh topologicalMesh( mesh );
+```
+
+ 3. Use OpenMesh to subdivide the geometry
+```cpp
+// Create OpenMesh subdivider, and process topological structure
+a.subdivider->attach( topologicalMesh );
+( *a.subdivider )( a.iteration );
+a.subdivider->detach();
+```
```
Comment (MathiasPaulin) [inline CLISubdivider/README.md:43]: Is it possible to simplify the writing to something like, e.g.
```cpp 
Subdivider subdivider( subdivisionType );
subdivider.attach( topologicalMesh );
subdivider(iterationNumber);
subdivider.detach();
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -10,7 +10,27 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
     set(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT False)
 endif()
 
+# Set default build type to Release
+if(NOT CMAKE_BUILD_TYPE)
+  messageInfo("No CMAKE_BUILD_TYPE, using default CMAKE_BUILD_TYPE=Release")
+  set(CMAKE_BUILD_TYPE "Release")
+endif()
+
```
Comment (MathiasPaulin) [inline CMakeLists.txt:18]: Put this before the test `if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)` and use `CMAKE_BUILD_TYPE` as suffix for installed directory (as for Radium-bundle). 
Keep the default install as `${CMAKE_CURRENT_BINARY_DIR}/installed-${CMAKE_CXX_COMPILER_ID}[-${CMAKE_BUILD_TYPE }]` with no suffix to for release install.

File: CMakeLists.txt [hunk 5]
```diff
@@ -10,7 +10,27 @@ if (CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
...
+# Be nice to visual studio
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+
+# Be nice and export compile commands by default, this is handy for clang-tidy
+# and for other tools.
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+# Append library and executable names with d in debug mode
+set(CMAKE_DEBUG_POSTFIX d)
```
Comment (MathiasPaulin) [inline CMakeLists.txt:30]: I do not like to suffix debug artifacts. Instead of this, there are installed in a specific folder.
When you have scripts that build-test-analyse-verify-relocate ... the executable or the lib, I found easier to change the base directory of the files manipulated into the script than adding a suffix to each file (with some that do not have suffix and some who have ...)
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -2,17 +2,36 @@
...
-This repository holds the front-end applications (GUI, command line) based on Radium Libraries.
+This repository holds front-end applications (GUI, command line) based on Radium Libraries
+ - Sandbox: Graphical frontend of the Radium Engine. This application aims at demonstrating the capabilities of the Radium Libraries. It can be extended using Radium plugins.
+ - CLISubdivider: Example of command line application. Loads obj files and run subdivision algorithms implemented with OpenMesh.
+ 
+To get more details about each application, checkout the Readme files in each application directory.
+
+# How to get these applications
+## Pre-compiled binaries
+Pre-compiled binaries can be compiled at https://github.com/STORM-IRIT/Radium-Releases/releases/.
```
Comment (MathiasPaulin) [inline README.md:13]:
```suggestion
Pre-compiled binaries can be downloaded from https://github.com/STORM-IRIT/Radium-Releases/releases/.
```

File: README.md [hunk 7]
```diff
@@ -2,17 +2,36 @@
...
+It includes:
```
Comment (MathiasPaulin) [inline README.md:14]:
```suggestion
The downloaded archive contains:
```

File: README.md [hunk 8]
```diff
@@ -2,17 +2,36 @@
...
+ - Radium libraries
+ - Radium applications
```
Comment (MathiasPaulin) [inline README.md:16]:
```suggestion
 - Radium exemple applications
     - A command line interface demo
     - A full featured graphical application
```

File: README.md [hunk 9]
```diff
@@ -2,17 +2,36 @@
...
+
+## Compilation instructions
 To compile this repository, you first need to compile the [Radium libraries](https://github.com/STORM-IRIT/Radium-Engine).
```
Comment (MathiasPaulin) [inline README.md:19]:
```suggestion
To compile this repository, you first need to compile and install the [Radium libraries](https://github.com/STORM-IRIT/Radium-Engine).
```

File: README.md [hunk 10]
```diff
@@ -2,17 +2,36 @@
...
+Then, configure this project using cmake. You need to give the location of Radium package by setting `Radium_DIR` (see more details in Radium Engine documentation: https://storm-irit.github.io/Radium-Engine/).
```
Comment (MathiasPaulin) [inline README.md:20]:
```suggestion
Then, configure this project using cmake. 
You need to tell cmake where to find the installed Radium package by setting the `Radium_DIR` variable to the right location (see more details in Radium Engine documentation: https://storm-irit.github.io/Radium-Engine/).
```

File: Sandbox/CMakeLists.txt [hunk 11]
```diff
@@ -1,12 +1,12 @@
...
+# Build Radium Sandbox
 cmake_minimum_required(VERSION 3.8)
 if(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} VERSION_GREATER "3.9")
     cmake_policy(SET CMP0071 NEW)
 endif()
 
 #-------------------------------------------------------------------------------
 # exampleApp executables setup
-project(main-app)
+project(Radium-Sandbox)
```
Comment (MathiasPaulin) [inline Sandbox/CMakeLists.txt:9]: Not fully convinced by this new name but I've no alternative proposal.

File: Sandbox/README.md [hunk 12]
```diff
@@ -0,0 +1,9 @@
+# Radium Sandbox
+
+This application has several purpose:
+ - It demonstrates standard graphical tools offered by the Radium-Engine: gizmos, picking, skeletal animation, keyframing, renderering, ...
+ - It offers a graphical front-end to load and use plugins easily. Just add your plugin to the Plugin path (see graphical user interface), its interface will appear in the command panel and you will be able to use it.
+
+Internally, we use this application for plugin development and testing of the Radium-Engine library.
```
Comment (MathiasPaulin) [inline Sandbox/README.md:7]: Well, this is only partly true (think about painty or Mara)...
```suggestion
Internally, we use this application as an integration and testing application of the Radium-Engine libraries.
```

**Comment by nmellado** [comment]
I have updated the PR (and force-pushed) according to your comments.

**Review by MathiasPaulin — APPROVED** [review]
